tests/unit/NetworkTest.js

```diff
@@ -0,0 +1,141 @@
+                    jest.advanceTimersByTime(1000);
+                    return waitForPromisesToResolve();
+                })
+                .then(() => {
+                    // Then we will eventually have 3 calls to chatList and 2 calls to Authenticate
+                    const callsToChatList = _.filter(HttpUtils.xhr.mock.calls, ([command, params]) => (
+                        command === 'Get' && params.returnValueList === 'chatList'
+                    ));
+                    const callsToAuthenticate = _.filter(HttpUtils.xhr.mock.calls, ([command]) => (
+                        command === 'Authenticate'
+                    ));
+
+                    expect(callsToChatList.length).toBe(3);
+                    expect(callsToAuthenticate.length).toBe(2);
+                });
+        });
+});
+
+test('consecutive API calls eventually succeed when authToken is expired', () => {
+    // Given a test user login and account ID
+    const TEST_USER_LOGIN = 'test@testguy.com';
+    const TEST_USER_ACCOUNT_ID = 1;
+
+    // When we sign in
+    return signInWithTestUser(TEST_USER_ACCOUNT_ID, TEST_USER_LOGIN)
+        .then(() => {
+            HttpUtils.xhr = jest.fn();
+            HttpUtils.xhr
+
+                // This will make the first call to API.Get() return with an expired session code
+                .mockImplementationOnce(() => Promise.resolve({
+                    jsonCode: 407,
+                }))
+
+                // The next 2 API calls will also fire and also return a 407
+                .mockImplementationOnce(() => Promise.resolve({
+                    jsonCode: 407,
+                }))
+
+                .mockImplementationOnce(() => Promise.resolve({
+                    jsonCode: 407,
+                }))
+
+                // The remaining requests should succeed
+                .mockImplementationOnce(() => Promise.resolve({
+                    jsonCode: 200,
+                    authToken: 'qwerty12345',
+                }))
+
+                .mockImplementation(() => Promise.resolve({
+                    jsonCode: 200,
+                }));
+
+            // And then make 3 API requests in quick succession with an expired authToken
+            API.Get({returnValueList: 'chatList'});
+            API.Get({returnValueList: 'personalDetailsList'});
+            API.Get({returnValueList: 'account'});
+            return waitForPromisesToResolve();
+        })
+        .then(() => {
+            // We should expect to see seven request be made in total. 3 Get requests that initially fail. Then the call
+            // to Authenticate. Followed by 3 requests to Get again.
+            const callsToGet = _.filter(HttpUtils.xhr.mock.calls, ([command]) => command === 'Get');
+            const callsToAuthenticate = _.filter(HttpUtils.xhr.mock.calls, ([command]) => command === 'Authenticate');
+            expect(callsToGet.length).toBe(6);
+            expect(callsToAuthenticate.length).toBe(1);
```

Comment:
These tests could be better. I just wanted to get something together quickly to offer some kind of insurance against  this breaking again. Mainly they could be improved by having the retried `Get` save the data in `Onyx` then we can look to see if the data has been set.